src/src/PluginStructs/P128_data_struct.cpp

```diff
@@ -363,28 +363,50 @@ bool P128_data_struct::plugin_write(struct EventStruct *event,
 
       _counter_mode_step = 0;
 
+      hex2rrggbb("000000");
+    /*CLEAN ALL PIXELS */
+      for (int i = 0; i < pixelCount; i++) {
+      Plugin_128_pixels->SetPixelColor(i, rrggbb);
+      }
       hex2rgb(str3);
 
       if (!str4.isEmpty()) { hex2rrggbb(str4); }
 
       speed = str5.isEmpty()
           ? defaultspeed
           : str5i;
+      ledi = str6.isEmpty()
+          ? int(1)
+          : str6i;
+      ledf = str7.isEmpty()
+          ? pixelCount
+          : str7i+1;
+          
     }
 
     else if (subCommand.equals(F("dualscan"))) {
       success = true;
       mode    = P128_modetype::Dualscan;
 
       _counter_mode_step = 0;
-
+ hex2rrggbb("000000");
+    /*CLEAN ALL PIXELS */
+      for (int i = 0; i < pixelCount; i++) {
+      Plugin_128_pixels->SetPixelColor(i, rrggbb);
+      }
       hex2rgb(str3);
 
       if (!str4.isEmpty()) { hex2rrggbb(str4); }
 
       speed = str5.isEmpty()
           ? defaultspeed
           : str5i;
+       ledi = str6.isEmpty()
+          ? int(1)
```

Comment:
An int doesn't need to be converted to an int.
Also there is a lot of code duplication here.
You already have parsed all those strX to an integer value strXi.
So why check them every time again?
Those int values will be 0 when the str they represent are empty and more importantly they will also be 0 when the string wasn't a numerical.

Reply:
Hi TD-er, I'm thinking about it...
I added two variables ledi and ledf (because I'm not very good programmer, and it's easy for me, not good for the code... I undestand that you say me)

If I have parsed the integer value in strXi, and it's "global" variable I can eliminate the two new variables ledi and ledf... And use strXi for the function calculations... I'm right?

Thanks for your help!

Reply:
OK, was just in my normal 'code review mode', not even realizing who was making the commit :)
I think the code was already "sub-optimal", so what we can do is first merge your code (as soon as Ton also looked at it) and then later try to have a proper look at this code as it seems to me it has way too much copy/paste code duplication in it.
But I think this does need a bit more attention to do it right.

One thing that could be done is to try and move all this code to a small function, but then we also need to make a small list of what's needed in what call.
So it might not be that simple to clean up.